Added dockerfile and compose file for docker deployment

alyssaholland99 opened the pull request:
Thought that more people would get use out of this tool if it was easy to deploy in a compose file. I've made a rudimentary Dockerfile and an example compose file, along with minor modifications to the setup script. 

I've made an image on my personal docker here (https://hub.docker.com/repository/docker/alyssaholland99/immich-tiktok-remover/general) but you're more than welcome to create your own image @mxc2 and swap it out in the compose file if you'd like :) 

All users need to do is edit the docker-compose.yml with their immich url and api key then run docker compose and the script should run every hour via a simple bash script.

alyssaholland99 committed:
Added dockerfile and compose file for docker deployment

tomribbens commented:
I wanted to try this, so I added your hosted image to my docker-compose.yml for my immich stack, and was surprised to see it being 6.4GB! I can see it's the pip3 install -r requirements.txt step that inflates the container size that much, I see.  

First thing I wonder, is if all those requirements are necessary, and if things could be done with less? But I'm certainly no expert on this, so let's assume this is all required.

Second option I see, would be to have the pip install step as part of the startup of the container, instead of in the image itself. I did a pip install on the requirements myself, and it seems that it only needs to download ~3GB of data. This would already improve the situation a lot.

If doing this last bit, it might be a good idea to have it download everything to a volume instead, so that it doesn't need to do that each time on startup.

tomribbens committed:
disable cache dir feature of pip to reduce image size

alyssaholland99 commented:
@tomribbens Yeah, I don't really like it being that big either - It is the requirements.txt that causes the inflation, I think it's the CUDA compatability that really bumps up the size. 

I'll try and move the pip install to the container start, rather than baking it in the image itself. Thanks for your suggestion :)

tomribbens commented:
@alyssaholland99 Have you looked at the PR I made after my comment? I think disabling the cache will already significantly reduce the size. 

I've since also learned that doing the pip install at runtime is something that goes against best practices. 

One idea I've had since, is that my downloaded TikTok videos always end with the same outro with the same audio. I know Plex & Jellyfin do credits detection based on audio, so maybe a rewrite of the code to do only audio detection might also help, but that's clearly out of scope for this PR.

alyssaholland99 commented:
@tomribbens Building now, thanks for the PR

alyssaholland99 commented:
@tomribbens That change for the image down to 3.4GB now :)

alyssaholland99 commented:
<img width="376" height="114" alt="image" src="https://github.com/user-attachments/assets/495c4e21-05b2-49f3-86ce-9685ae2e14b6" />

Managed to reduce the number of recent vulnerabilities with the image by a lot.

Also reduced the image size by ~300MB

alyssaholland99 committed:
Added a stable-lite image to Docker

Added stable-lite image to drone run

Added --avoid-image-recognition to the make_lite

Tidy up

Added args to the docker image

ARG > ENV

Updated docker compose example

Updated README to reflect 1.137 changes

Change drone repository to drone secret

Updated README to reflect new environment variables

Fixed repo link

Clean up drone file

Added stable-lite comment

Fixed image tag

Update README to reflect stable-lite comment

mxc2 commented:
Nice work, I'll give it a test soon :) 

On the topic of image size, would be interesting to know, how many people actually use the image recognition part of the script. I personally disabled it on my server and just run the file name checks, since it's faster and I know that I don't have any files that are in the style of Tiktok file names. Creating a version without the image recognition would reduce the file size alot.

alyssaholland99 commented:
I personally use it but we could always have a couple of docker images for people that want to use the image recognition part of the tool and those that don't?

mxc2 commented:
Ehh, probably not worth it atm.

Is it possible to use customization options like "--file-name-length" with the docker version?

alyssaholland99 commented:
Yea, it's currently baked into the image atm so it'd have to be rebuilt but we could have them as ENVs? Just been doing some testing on a 'lite' image without the image recognition and the size is about 430MB which is much better then the 3GB including the image recognition.

I'll keep playing about with it when I can and see if I can get a 'stable' and a 'stable-lite' version configured in the drone config

mxc2 commented:
Ye having them as ENVs would be good.

Damn that is a big downsize in the size. If it isn't much more work, then I stand corrected, lite image would be good to have.

mxc2 commented:
Hey, just incase wanted to ask, any updates on the 'stable' and 'stable-lite'?

alyssaholland99 commented:
Hey, sorry I haven't had time at the moment. I'll try get to it tomorrow :)

mxc2 commented:
No worries and no need to hurry, was just wondering, there's no pressure to get it done fast :)

alyssaholland99 commented:
@mxc2 https://github.com/alyssaholland99/immich-tiktok-remover/pull/4/files Got a PR up there, let me know if you can see any issues with it :) I'll play about with it later and see if I can get it a bit cleaner than that but it should be done tomorrow :) 

Drone run: https://drone.alyssaserver.co.uk/alyssaholland99/immich-tiktok-remover/95

Image size with EasyOCR: 3.1GB
Image size without EasyOCR: 102MB
https://hub.docker.com/r/alyssaholland99/immich-tiktok-remover

mxc2 commented:
Hey, just incase wanted to check, does this build support customization options through ENVs?

alyssaholland99 commented:
Oh, that's a good shout - I'll try add those next week :)

alyssaholland99 commented:
Added the environment variables. Seems to work for me but would be interested how you get on with it?

Going to clean up the drone file over the weekend to make it more readable too

alyssaholland99 commented:
Rewrote the drone file to allow build on tagging

https://drone.alyssaserver.co.uk/alyssaholland99/immich-tiktok-remover/140
^^ Tagging run 

https://drone.alyssaserver.co.uk/alyssaholland99/immich-tiktok-remover/142
^^ This PR's most recent drone run

Commit:
Added a way to test Docker images with a standalone Immich instance

Added a way to test the image in drone

Make the test only run after the build

Fixed drone steps to only run when needed

Changed ; for &&

Added sleep

Removed dependancy for testing

Removed templates test

Added back templates

docker ps for testing

Added a script to test itr

Drone should now work?

Added documentation to testing

Removed build for testing

Stopped the testing running in drone due to env issues

Updated readme for Drone

Made the cleanup script have to output when running locally

Fixed lite version

Fixed tz

Added a confirmation screen to warn people not to run in a live environment

Added an option to run the lite version

Allowed tagging of image with lite

Added tagging to lite setup script

More lite changes

Fixed drone

Removed videos & Drone cleanup

Added video directories to gitignore

alyssaholland99 commented:
I've added integration tests for the Dockerhub images to this PR too

alyssaholland99 commented:
<img width="1657" height="812" alt="Screenshot from 2025-08-17 19-47-46" src="https://github.com/user-attachments/assets/2e0409f4-bfdc-45a3-b596-ee9b866956c6" />

Output of the run_tesh.sh script on the stable tag. The stable-lite tag doesn't seem to be working at the moment. Will diagnose tomorrow but I reckon it's an issue with the image and not with the test

alyssaholland99 commented:
<img width="1657" height="812" alt="Screenshot from 2025-08-18 04-39-28" src="https://github.com/user-attachments/assets/f0af7173-3122-437a-ae68-8959735bba9a" />

Managed to fix the stable-lite image and ran the same test, got the same result so I think both stable tags work :)

alyssaholland99 commented:
I think this is ready to review now @mxc2 - Let me know of any issues you find / features needed :)

alyssaholland99 commented:
[Screencast from 2025-08-18 05-45-15.webm](https://github.com/user-attachments/assets/4ff0b569-03b6-49bf-a68a-4fa53b0a39ab)

Added a demo of the tests. It's very boring to watch so I'd skip though most of it if you want to watch it

mxc2 commented:
Sweet, I'll try it out tomorrow or day after :)

mxc2 commented:
Hey, so sorry for the longer wait, been busy irl. 

I've been thinking about the docker tests. I think it would be better to add an additional param to the Python code, that just does a dry run on the user's Immich system.

My main concern is that adding videos to Git is not really recommended. It massively increases the size of the repo without adding functionality to the main code. Additionally, if a user is looking for a way to remove Tiktok videos from Immich, then most likely they already have a Immich server setup and it would be quite easy for them to do a dry run on their own system.

I can code the dry run logic into Python, then add the param to Docker and I believe then it would be okay to remove the testing logic?

alyssaholland99 commented:
Sorry, I've been busy for the past few weeks. To be fair I created the tests mainly for development because I like having an environment completely seperate from my Immich install. #

Yeah, not exactly a great idea to put videos into git so I'll get rid of those but are you good with the testing script being kept? I'll also clean up the drone file to remove that testing for the time being :) 

I think the parameters are working in the docker container too; I'll do a little bit more testing when I clean up the branch

alyssaholland99 commented:
^^ Just cleaned up the commits too :)

mxc2 commented:
Sweet, good job! Sorry for the long PR lol, I'll merge it now.

alyssaholland99 commented:
No worries, glad you suggested some features to go along with it :)
